mycpp/mylib_leaky.cc

```diff
@@ -11,6 +11,12 @@
 
 Str* kEmptyString = new Str("", 0);
 
+
+Str* AllocStr(strip_result str) {
+  return new Str(str.data_, str.len_);
```

Comment:
Correct, the old routines returned slices (as noted by the `cstring-NOTE` and the `TODO` you mentioned) so this shouldn't cause regressions.  I wrote the `AllocStr` function like that precisely _because_ the old routines returned slices;  I didn't want to alter the behavior without good reason.

That said, it's probably a good idea to return copies since the GC scheme doesn't give us a choice anyways.